sonaranalyzer-dotnet/rspec/cs/S5547_c#.html

```diff
@@ -2,17 +2,30 @@
 are not vulnerable to well-known attacks like brute force attacks for example. </p>
 <p>It is recommended to use only cipher algorithms intensively tested and promoted by the cryptographic community.</p>
 <h2>Noncompliant Code Example</h2>
+<p>For <a href="https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography?view=netcore-3.1">System.Security.Cryptography</a> library,
+these old cryptographic algorithms should no longer be used for any reason:</p>
 <pre>
 var tripleDES1 = new TripleDESCryptoServiceProvider(); // Noncompliant: Triple DES is vulnerable to meet-in-the-middle attack
 
 var simpleDES = new DESCryptoServiceProvider(); // Noncompliant: DES works with 56-bit keys allow attacks via exhaustive search
 
 var RC2 = new RC2CryptoServiceProvider(); // Noncompliant: RC2 is vulnerable to a related-key attack
 </pre>
+<p>For Bouncycastle library, <a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-1000339">AESFastEngine has a side channel leak</a>, it
+is possible to gain information about the key used to initialize the cipher:</p>
+<pre>
+AesFastEngine aesfast = new AesFastEngine(); // Noncompliant
+</pre>
 <h2>Compliant Solution</h2>
+<p>For <a href="https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography?view=netcore-3.1">System.Security.Cryptography</a> library,
+it's recommended to use <code>AesCryptoServiceProvider</code>:</p>
 <pre>
 var AES = new AesCryptoServiceProvider(); // Compliant
 </pre>
+<p>For Bouncycastle library, it's recommended to use <code>AESEngine</code>:</p>
+<pre>
+var AES = new AESEngine(); // Compliant
+</pre>
```

Comment:
Changes in this RSPEC do not reflect current behaviour of analyzer. We should keep previous file version until #3330 is fixed